core/API/Request.php

```diff
@@ -701,4 +702,47 @@ private function shouldDisablePostProcessing()
 
         return $shouldDisable;
     }
+
+    public static function isCurrentPeriodProfilable($idSite = null, $period = null, $date = null, $segment = null)
+    {
+        if (self::isProfilableCheckDisabledInTests()) {
+            return true;
+        }
+
+        $idSite = $idSite ?: Common::getRequestVar('idSite', $default = false);
+        $period = $period ?: Common::getRequestVar('period', $default = false);
```

Comment:
I'm thinking we'd want to convert a 'day' period to at least `week`. This way, if some days have profilables and some don't the experience is a bit more consistent and less unpredictable? This can happen eg when people can opt in to cookies or not. Some days maybe all visitors opt out vs some other days some opt in etc.

Reply:
So only disable reports/etc. for a day if the entire week has enough profilable data? Sounds doable.

Reply:
👍

Reply:
We'd basically simply replace day period with week